R/derive_joined.R

```diff
@@ -371,52 +423,198 @@
 #'
-#' # derive first "++" day after "0" where all results in between are "+" or "++"
+#' - The `join_type = "after"` here essentially acts as a lag to join variables from
+#'   the next occurring record, and `mode = "first"` selects the first of these.
+#' @code
 #' derive_vars_joined(
 #'   myd,
 #'   dataset_add = myd,
-#'   by_vars = exprs(subj),
-#'   order = exprs(day),
+#'   by_vars = exprs(STUDYID, USUBJID),
+#'   order = exprs(ADY),
+#'   mode = "first",
+#'   new_vars = exprs(NEXTVAL = AVAL),
+#'   join_vars = exprs(AVAL),
+#'   join_type = "after"
+#' ) %>%
+#'   select(USUBJID, ADY, AVAL, NEXTVAL)
+#'
+#' @caption Join records after a condition occurring in consecutive visits
+#'   (`tmp_obs_nr_var`, `join_type` and `filter_join`)
+#' @info Find the last occurring value on any of the next 3 unique visit days.
+#'
+#' - The `tmp_obs_nr_var` argument can be useful as shown here to help pick out
+#'   records happening before or after with respect to `order`, as you can see
+#'   in the `filter_join`.
+#' @code
+#' derive_vars_joined(
+#'   myd,
+#'   dataset_add = myd,
+#'   by_vars = exprs(STUDYID, USUBJID),
+#'   order = exprs(ADY),
 #'   mode = "last",
-#'   new_vars = exprs(next_plus_day = day),
-#'   join_vars = exprs(val),
+#'   new_vars = exprs(NEXTVAL = AVAL),
+#'   tmp_obs_nr_var = tmp_obs_nr,
+#'   join_vars = exprs(AVAL),
 #'   join_type = "after",
-#'   first_cond_upper = val.join == "++",
-#'   filter_join = val == "0" & all(val.join %in% c("+", "++"))
+#'   filter_join = tmp_obs_nr + 3 >= tmp_obs_nr.join
+#' ) %>%
+#'   select(USUBJID, ADY, AVAL, NEXTVAL)
+#'
+#' @caption Derive period variables (`APERIOD`, `APERSDT`, `APEREDT`)
+#' @info Create a period reference dataset from `ADSL` and join this with `ADAE`
+#'   to identify within which period each AE occurred.
+#' @code
+#' adsl <- tribble(
+#'   ~USUBJID, ~AP01SDT,     ~AP01EDT,     ~AP02SDT,     ~AP02EDT,
+#'   "1",      "2021-01-04", "2021-02-06", "2021-02-07", "2021-03-07",
+#'   "2",      "2021-02-02", "2021-03-02", "2021-03-03", "2021-04-01"
+#' ) %>%
+#'   mutate(across(ends_with("DT"), ymd)) %>%
+#'   mutate(STUDYID = "AB42")
+#'
+#' period_ref <- create_period_dataset(
+#'   adsl,
+#'   new_vars = exprs(APERSDT = APxxSDT, APEREDT = APxxEDT)
 #' )
+#'
+#' period_ref
+#'
+#' adae <- tribble(
+#'   ~USUBJID, ~ASTDT,
+#'   "1",      "2021-01-01",
+#'   "1",      "2021-01-05",
+#'   "1",      "2021-02-05",
+#'   "1",      "2021-03-05",
+#'   "1",      "2021-04-05",
+#'   "2",      "2021-02-15",
+#' ) %>%
+#'   mutate(
+#'     ASTDT = ymd(ASTDT),
+#'     STUDYID = "AB42"
+#'   )
+#'
+#' derive_vars_joined(
+#'   adae,
+#'   dataset_add = period_ref,
+#'   by_vars = exprs(STUDYID, USUBJID),
+#'   join_vars = exprs(APERSDT, APEREDT),
+#'   join_type = "all",
+#'   filter_join = APERSDT <= ASTDT & ASTDT <= APEREDT
+#' ) %>%
+#'   select(USUBJID, ASTDT, APERSDT, APEREDT, APERIOD)
+#'
+#' @caption Further examples
```

Comment:
Love this!! But was thinking - should we have something at the top of these examples explaining to users when you might want to use join versus merge - it is very confusing.  Linking to document is nice but just stating in a pithy way would be nice!!